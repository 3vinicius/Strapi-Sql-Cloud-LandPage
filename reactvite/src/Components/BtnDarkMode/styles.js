import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${() => css`
    height: 80px;
    width: 50px;
    background-color: red;
    padding: 10px;
    border-radius: 25px;
    margin: 10px;
    cursor: pointer;
  `}
`;
