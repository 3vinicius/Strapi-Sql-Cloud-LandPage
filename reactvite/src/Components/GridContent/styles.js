import styled, { css } from 'styled-components';
import darkMode from '../../Utils/stylesUtils';

export const Container = styled.div`
  ${({ theme, colorDark }) => css`
    text-align: center;
    max-width: 58rem;
    margin: auto;
    font-size: ${theme.fonts.sizes.medium};
    ${darkMode(colorDark)}
    ;
  `}
`;

export const Html = styled.div`
  ${({ theme }) => css`
    margin: ${theme.spacings.xhuge}
  `}
`;
