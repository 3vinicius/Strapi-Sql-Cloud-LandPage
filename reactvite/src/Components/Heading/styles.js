import styled, { css } from 'styled-components';

// sizes: {
//   xsmall: '0.8rem',
//   lsmall: '1.6rem',
//   medium: '2.4rem',
//   large: '3.2rem',
//   xlarge: '4.0rem',
//   xxlarge: '4.8rem',
//   huge: '5.6rem',
//   xhuge: '6.4rem',
// },
const titleSize = {
  small: (theme) => css`
  font-size: ${theme.fonts.sizes.medium} ;
  `,
  medium: (theme) => css`
    font-size: ${theme.fonts.sizes.large} ;
  `,
  big: (theme) => css`
  font-size: ${theme.fonts.sizes.xhuge} ;
  ${mediaFonte(theme)}

`,
};

const mediaFonte = (theme) => css`@media ${theme.myMedias.letMedium} {
  font-size: ${theme.fonts.sizes.large};
}`;

export const Title = styled.h1`
  ${({ theme, colorDark, size, uppercase }) => css`
    color: ${colorDark ? theme.colors.colorWhite : theme.colors.primaryColor};
    ${titleSize[size](theme)}
    text-transform: ${uppercase ? 'uppercase' : 'none'};
  `}
`;
