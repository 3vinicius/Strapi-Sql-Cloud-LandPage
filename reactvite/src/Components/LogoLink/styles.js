/* eslint-disable no-unused-vars */
import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`;
  `}
  display: flex;
  align-items: center;
  > img {
    max-width: 8rem;
    max-height: 8rem;
  }
`;
