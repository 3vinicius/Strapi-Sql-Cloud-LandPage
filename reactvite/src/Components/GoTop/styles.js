import styled, { css } from 'styled-components';
import { theme } from '../../styles/theme';

export const Container = styled.a`
  ${() => css``}
  position: fixed;
  background: ${theme.colors.primaryColor};
  color: ${theme.colors.colorWhite};
  display:flex;
  align-items: center;
  justify-content: center;
  height:4rem;
  width: 4rem;
  bottom: 2rem;
  right: 2rem;
  z-index: 6;
  opacity: 0.7;
  transition: all 0.5s ease-out  ;
  border: 1px solid #060a18;
  box-shadow: 0px 0px 5px 2px #00000054;


  &:hover{
    opacity: 1;
  }
`;
