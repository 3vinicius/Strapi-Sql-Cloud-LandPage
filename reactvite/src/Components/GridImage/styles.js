import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
  padding: 200px
  ${TextComponent}{
    margin-bottom: ${theme.spacings.xhuge};
    ;
  }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large}
  `}
`;

export const GridElement = styled.div`
  ${() => css`
  overflow: hidden;
  `}
`;

export const Imagem = styled.img`
  ${() => css`
    width:100%;
    height: 100%;
    transition: all 300ms ease-in-out;

    &:hover{
      transform: scale(1.2)
    }
  `}
`;
