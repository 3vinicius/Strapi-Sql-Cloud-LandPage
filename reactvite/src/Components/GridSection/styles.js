import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme, colorDark }) => css`
  ${TextComponent}{
    margin-bottom: ${theme.spacings.xhuge};
    background: ${
      colorDark ? theme.colors.primaryColor : theme.colors.colorWhite
    };
  }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large}
  `}
`;

export const GridElement = styled.div`
  ${() => css`
  ${Title}{
    position: relative;
    left: 5rem;
  }
  ${Title}::before{
    counter-increment: grid-counter;
    content: counter(grid-counter);
    position: absolute;
    font-size: 7rem;
    top: -2rem;
    left: -5rem;
    transform: rotate(5deg);
  }
  `}
`;
