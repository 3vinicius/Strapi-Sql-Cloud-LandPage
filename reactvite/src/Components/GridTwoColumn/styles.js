import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';
import { Container as Texts } from '../TextComponent/styles';
import darkMode from '../../Utils/stylesUtils.js';

// primaryColor: string;
//         secondaryColor: string;
//         colorWhite: string;

export const Container = styled.div`
  ${({ theme, background }) => css`

    display:grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.spacings.large};
    @media(max-width: 850px){
      grid-template-columns: 1fr;
      text-align: center;
    }


    ${Texts}{
      ${darkMode(background)}
    }
    ${Title}{
      margin-bottom: ${theme.spacings.large};
      ${darkMode(background)}
    }

  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
  @media ${theme.myMedias.letMedium}{
      margin-bottom: ${theme.spacings.large};
    }
  `}
`;

export const ImgContainer = styled.div`
  ${({ theme }) => css``}
`;

export const Img = styled.img`
  ${({ theme }) => css`
  width: 100%;
  min-width: 200px;
  overflow: hidden;
  @media(max-width: 850px){
      width: 400px;
    }
  @media ${theme.myMedias.letMedium}{
    width:300px;
  }
  `}
`;
