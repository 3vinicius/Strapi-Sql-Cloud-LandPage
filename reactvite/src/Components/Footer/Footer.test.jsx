import { screen } from '@testing-library/react';
import { Footer } from '.';
import { renderTheme } from '../../styles/render-theme';

const html = `<h1>Create by Vinicus Amorim</h1>`;

describe('<Footer />', () => {
  it('Should render <Footer />', () => {
    const { container } = renderTheme(<Footer html={html} />);
    expect(screen.getAllByRole('heading', { name: 'Create by Vinicus Amorim' }))
      .toBeInTheDocument;
    expect(container).toMatchSnapshot();
  });
});
