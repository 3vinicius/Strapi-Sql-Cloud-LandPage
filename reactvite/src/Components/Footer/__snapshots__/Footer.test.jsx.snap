// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> Should render <Footer /> 1`] = `
.c3 {
  font-size: 2.4rem;
  color: #0A1128;
}

.c2 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c0 {
  text-align: center;
  border: 0.1rem solid #DDDDDD;
}

.c0 a {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 .c1 {
  padding-top: 0px;
  padding-bottom: 0px;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1 c2"
    >
      <div>
        <p
          class="c3"
        >
          <h1>
            Create by Vinicus Amorim
          </h1>
        </p>
      </div>
    </div>
  </div>
</div>
`;
